tensorflow_datasets/audio/audioset.py

```diff
@@ -0,0 +1,104 @@
+booktitle  = {Proc. IEEE ICASSP 2017},
+address = {New Orleans, LA}
+}
+"""
+
+_DESCRIPTION = """
+The AudioSet dataset is a large-scale collection of human-labeled 10-second sound clips drawn from YouTube videos. 
+To collect all our data we worked with human annotators who verified the presence of sounds they heard within YouTube segments. 
+To nominate segments for annotation, we relied on YouTube metadata and content-based search.
+"""
+
+_DOWNLOAD_LINK = 'https://storage.googleapis.com/bme590/william/audioset.tar.gz'
+
+class Audioset(tfds.core.GeneratorBasedBuilder):
+  """This dataset takes in a manual directory of 10-second .mp3 files taken from Youtube.
+  It yields Audio using pydub AudioSegment and corresponding labels
+  according to the CLASS_LABELS_CSV provided by Audioset.
+  """
+
+  VERSION = tfds.core.Version('0.1.0')
+  MANUAL_DOWNLOAD_INSTRUCTIONS = 'give bucket path gs://bme590/william/manual'
+  def _info(self):
+    return tfds.core.DatasetInfo(
+      builder=self,
+      description=_DESCRIPTION,
+      features=tfds.features.FeaturesDict({
+        'audio': tfds.features.Audio(file_format=None, sample_rate=48000),
+        'label': tfds.features.Tensor(shape=(None,), dtype=tf.int32)
+      }),
+      supervised_keys=('audio','label'),
+      homepage='https://research.google.com/audioset/index.html',
+      citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    dl_manager._register_checksums = True
+    data_dir = dl_manager.download_and_extract(_DOWNLOAD_LINK)
+    return [
+      tfds.core.SplitGenerator(
+        name=tfds.Split.TRAIN,
+        gen_kwargs={
+          'data_dir': data_dir
+        },
+      ),
+    ]
+
+  def _generate_examples(self, data_dir=None):
+    """Yields examples."""
+    trimmed_audio_path = os.path.join(data_dir, 'trimmed_audio')
+    id2label_path = os.path.join(data_dir, 'id2label.json')
+    my_files = tf.io.gfile.listdir(trimmed_audio_path) #gets list of files in folder given
+    with tf.io.gfile.GFile(id2label_path, "r") as read_file: #uses json to match ids to labels
+      datas = json.load(read_file)
+    for f in my_files:
+      try:
+        label_list = []
+        filepath = os.path.join(trimmed_audio_path, f)
+        with tf.io.gfile.GFile(filepath, "rb") as read_file:
+          errorcheck = AudioSegment.from_file(read_file) #ignores bad mp3 files
+        ids = f.replace(".mp3","")
+        for x in datas[ids]:
+          for y in x:
+            label_list.append(y) #gets multiple labels index that apply to audio in a list in label_list
+          label_tensor = np.zeros(527, dtype=np.int32)
+          for i in label_list:
+            label_tensor[i] = 1 # creates a tensor of zeros with ones at indices that indicate positive for a label
+          break
+        yield f, {
+          'audio': filepath,
+          'label': label_tensor,
+        }
+      except:
+        print(filepath)
+        pass
```

Comment:
Better msg for corrupted data (See https://www.tensorflow.org/datasets/add_dataset#corrupted_data).

Reply:
I've removed the print line and added an explanation in the dataset description for missing/corrupted examples.